src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c

```diff
@@ -325,6 +330,9 @@ static void tvhdhomerun_device_open_pid(tvhdhomerun_frontend_t *hfe, int pid) {
 
   tvhdebug("tvhdhomerun", "current pfilter: %s", pfilter);
 
+  /* make sure the pid maps to a max of 0x1FFF, API will reject the call otherwise */
+  pid = (pid & 0x1FFF);
```

Comment:
It will be probably better to show a warning here if pid is above 0x1fff and/or simply return.